### PR TITLE
build: use `GNUInstallDirs` and improve packaging

### PR DESCRIPTION
**Pull request by c-dilks**
### Briefly, what does this PR introduce?

- Use `GNUInstallDirs` to set the installation tree, instead of doing it "by hand"
- Install `cmake` files in `lib/cmake/IRT` instead of in `lib/IRT`
- Add public-facing variables `IRT_LIBRARY_DIR` and `IRT_INCLUDE_DIR`
- This requires updates in EICRecon: https://github.com/eic/EICrecon/pull/546

#### Installation tree before this PR:
```
CMAKE_INSTALL_PREFIX
├── include
│   └── IRT
│       ├── BitMask.h
│       ├── ChargedParticle.h
│       ├── ChargedParticleStep.h
│       ├── CherenkovDetectorCollection.h
│       ├── CherenkovDetector.h
│       ├── CherenkovEvent.h
│       ├── CherenkovMirror.h
│       ├── CherenkovPhotonDetector.h
│       ├── CherenkovPID.h
│       ├── CherenkovRadiator.h
│       ├── G4Object.h
│       ├── IRT.h
│       ├── IRTSolution.h
│       ├── OpticalBoundary.h
│       ├── OpticalPhoton.h
│       ├── ParametricSurface.h
│       ├── RadiatorHistory.h
│       ├── ReflectionPoint.h
│       ├── RefractionPoint.h
│       ├── SinglePDF.h
│       └── TransientParticle.h
└── lib
    ├── IRT
    │   ├── IRTConfig.cmake
    │   ├── IRTConfigVersion.cmake
    │   ├── IRTTargets.cmake
    │   └── IRTTargets-noconfig.cmake
    ├── libDELPHES_rdict.pcm
    ├── libDELPHES.rootmap
    ├── libDELPHES.so
    ├── libIRT_rdict.pcm
    ├── libIRT.rootmap
    └── libIRT.so
```

#### After this PR:
```
CMAKE_INSTALL_PREFIX
├── include
│   └── IRT
│       ├── BitMask.h
│       ├── ChargedParticle.h
│       ├── ChargedParticleStep.h
│       ├── CherenkovDetectorCollection.h
│       ├── CherenkovDetector.h
│       ├── CherenkovEvent.h
│       ├── CherenkovMirror.h
│       ├── CherenkovPhotonDetector.h
│       ├── CherenkovPID.h
│       ├── CherenkovRadiator.h
│       ├── G4Object.h
│       ├── IRT.h
│       ├── IRTSolution.h
│       ├── OpticalBoundary.h
│       ├── OpticalPhoton.h
│       ├── ParametricSurface.h
│       ├── RadiatorHistory.h
│       ├── ReflectionPoint.h
│       ├── RefractionPoint.h
│       ├── SinglePDF.h
│       └── TransientParticle.h
└── lib
    ├── cmake
    │   └── IRT
    │       ├── IRTConfig.cmake
    │       ├── IRTConfigVersion.cmake
    │       ├── IRTTargets.cmake
    │       └── IRTTargets-noconfig.cmake
    ├── libDELPHES_rdict.pcm
    ├── libDELPHES.rootmap
    ├── libDELPHES.so
    ├── libIRT_rdict.pcm
    ├── libIRT.rootmap
    └── libIRT.so
```

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [x] Other: refactor the build config

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
Yes, see above PR on EICrecon. This also might break Juggler's usage (on a dev branch, not `main`).
This also might impact @alexander-kiselev's usage

### Does this PR change default behavior?
No, but hopefully it solves our various build/linking/`cling`/etc. problems in EICrecon...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,47 +87,48 @@ endif()
...
-install(TARGETS ${IRT_LIB}
-  EXPORT ${PROJECT_NAME}Targets
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
-  RUNTIME DESTINATION bin
-  INCLUDES DESTINATION include/IRT
+install(TARGETS ${CMAKE_PROJECT_NAME}
+  EXPORT ${CMAKE_PROJECT_NAME}Targets
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}
```
Comment (veprbl) [inline CMakeLists.txt:95]: Extra `${CMAKE_PROJECT_NAME}` sub-directory is quite non-standard.
```suggestion
  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -89,47 +87,48 @@ endif()
...
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_PROJECT_NAME}
 )
 install(FILES
   ${IRT_ROOTMAP}
-  DESTINATION lib
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME}
```
Comment (veprbl) [inline CMakeLists.txt:101]:
```suggestion
  DESTINATION ${CMAKE_INSTALL_LIBDIR}
```

File: cmake/IRTConfig.cmake.in [hunk 3]
```diff
@@ -1,3 +1,5 @@
 @PACKAGE_INIT@
 include("@PACKAGE_TARGETS_INSTALL_PATH@")
+set_and_check(IRT_INCLUDE_DIR "@PACKAGE_CMAKE_INSTALL_INCLUDEDIR@")
+set_and_check(IRT_LIBRARY_DIR "@PACKAGE_CMAKE_INSTALL_LIBDIR@/IRT")
```
Comment (veprbl) [inline cmake/IRTConfig.cmake.in:4]:
```suggestion
set_and_check(IRT_LIBRARY_DIR "@PACKAGE_CMAKE_INSTALL_LIBDIR@")
```

File: delphes/CMakeLists.txt [hunk 4]
```diff
@@ -28,6 +28,6 @@ TARGET_LINK_LIBRARIES(${DELPHES_LIB} ${ROOT_LIBRARIES} EG )
 
 #----------------------------------------------------------------------------
 
-INSTALL(FILES   ${DELPHES_ROOTMAP} DESTINATION lib)
-install(TARGETS ${DELPHES_LIB}     DESTINATION lib)
+INSTALL(FILES   ${DELPHES_ROOTMAP} DESTINATION ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME})
+install(TARGETS ${DELPHES_LIB}     DESTINATION ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME})
```
Comment (veprbl) [inline delphes/CMakeLists.txt:32]:
```suggestion
INSTALL(FILES   ${DELPHES_ROOTMAP} DESTINATION ${CMAKE_INSTALL_LIBDIR})
install(TARGETS ${DELPHES_LIB}     DESTINATION ${CMAKE_INSTALL_LIBDIR})
```

**Comment by c-dilks** [comment]
Tested IRT PID in EICrecon, still works fine.

File: CMakeLists.txt [hunk 5]
```diff
@@ -62,13 +57,16 @@ set(IRT_SRC
...
+  ${CMAKE_CURRENT_BINARY_DIR}/lib${CMAKE_PROJECT_NAME}.rootmap
+)
 root_generate_dictionary(G__IRT ${HEADERS} LINKDEF irtLinkDef.h)
 
-add_library(${IRT_LIB} SHARED ${IRT_SRC} )
-target_link_libraries(${IRT_LIB} ROOT::Core)
-target_compile_options(${IRT_LIB} PRIVATE -Wall -Wno-misleading-indentation)
-target_include_directories(${IRT_LIB} PUBLIC
+add_library(${CMAKE_PROJECT_NAME} SHARED ${IRT_SRC} )
+target_link_libraries(${CMAKE_PROJECT_NAME} ROOT::Core)
```
Comment (veprbl) [inline CMakeLists.txt:67]: Not related but TVector3 comes from libPhysics
```suggestion
target_link_libraries(${CMAKE_PROJECT_NAME} ROOT::Core ROOT::Physics)
```
```
Undefined symbols for architecture x86_64:
  "TVector3::Rotate(double, TVector3 const&)", referenced from:
      SphericalSurface::_Clone(double, TVector3 const&) const in SphericalSurface.cc.o
      FlatSurface::_Clone(double, TVector3 const&) const in FlatSurface.cc.o
      IRT::Transport(TVector3 const&, TVector3 const&) in IRT.cc.o
      IRT::Solve(TVector3 const&, TVector3 const&, double const*, TVector3 const&, bool, IRTSolution const*) in IRT.cc.o
      FlatSurface::_Clone(double, TVector3 const&) const in G__IRT.cxx.o
      SphericalSurface::_Clone(double, TVector3 const&) const in G__IRT.cxx.o
  "TVector3::Phi() const", referenced from:
      IRT::Solve(TVector3 const&, TVector3 const&, double const*, TVector3 const&, bool, IRTSolution const*) in IRT.cc.o
  "TVector3::Unit() const", referenced from:
      SphericalSurface::GetNormal(TVector3 const&) const in SphericalSurface.cc.o
      IRT::Transport(TVector3 const&, TVector3 const&) in IRT.cc.o
      IRT::Solve(TVector3 const&, TVector3 const&, double const*, TVector3 const&, bool, IRTSolution const*) in IRT.cc.o
      ChargedParticle::PIDReconstruction(CherenkovPID&) in ChargedParticle.cc.o
      SphericalSurface::GetNormal(TVector3 const&) const in G__IRT.cxx.o
  "TVector3::Theta() const", referenced from:
      IRT::Solve(TVector3 const&, TVector3 const&, double const*, TVector3 const&, bool, IRTSolution const*) in IRT.cc.o
  "vtable for TVector3", referenced from:
      SphericalSurface::GetSpacePoint(double, double) const in SphericalSurface.cc.o
      ParametricSurface::GetNormal() const in SphericalSurface.cc.o
      SphericalSurface::_Clone(double, TVector3 const&) const in SphericalSurface.cc.o
      FlatSurface::GetNormal() const in FlatSurface.cc.o
      FlatSurface::FlatSurface(FlatSurface const&) in FlatSurface.cc.o
      IRT::Transport(TVector3 const&, TVector3 const&) in IRT.cc.o
      IRT::Solve(TVector3 const&, TVector3 const&, TVector3 const&, TVector3 const&, bool, IRTSolution const*) in IRT.cc.o
      ...
```

**Review by veprbl — APPROVED** [review]
LGTM

**Pull request merged**